### PR TITLE
chore: merging release v0.29 into master

### PR DESCRIPTION
**Pull request by Ivansete-status**
## Changes
- Update the changelog
- bump the `nim-libp2p`. This, amongst other things, adds log enhancements that will help to debug how the peer score evolves.
- The changes in `node/waku_node.nim` and `waku_relay/protocol.nim` come from cherry-picking the following commit: a7d2dfd0aabf70e71bd0eb804d45e086a5d297e7

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,66 @@
+## v0.29.0 (2024-06-11)
+
+## What's Changed
+
+Notes:
+
+* Start deprecate named sharding ([PR](https://github.com/waku-org/nwaku/pull/2723))
```
Comment (gabrielmer) [inline CHANGELOG.md:7]: I think we should add a more clear deprecation notice such as

```suggestion
* Named sharding will be deprecated in favor of static sharding. Topics in formats other than `/waku/2/rs/<cluster>/<shard>` will stop being supported starting from `v0.31.0`
```

**Review by gabrielmer — APPROVED** [review]
LGTM, thanks so much!

**Review by NagyZoltanPeter — APPROVED** [review]
LGTM

**Review by darshankabariya — APPROVED** [review]
LGTM !

**Pull request merged**